test/CalendarDate.test.ts

```diff
@@ -19,12 +19,14 @@ describe('CalendarDate', () => {
               // Assert
               expect(
                 () => new CalendarDate(year, undefinedAsNumber, undefinedAsNumber),
-              ).toThrowError(`CalendarDate Validation Error: Input [ ${year} ] is not valid.`);
+              ).toThrowError(
+                `CalendarDate Validation Error: Input [ ${year} ] of type [ number ] is not valid.`,
```

Comment:
```suggestion
              ).toThrowError(
                `CalendarDate Validation Error: Input [ ${year} ] of type [ number ] is not valid. Provided type: ${typeof year}`,
```

Maybe add the type of the provided input to the error?